cou/apps/app.py

```diff
@@ -394,7 +407,7 @@ def upgrade_plan(self, target: OpenStackRelease) -> list[Optional[UpgradeStep]]:
         :return: Plan that will add upgrade as sub steps.
         :rtype: list[Optional[UpgradeStep]]
         """
-        os_origin_apt_sources = self.os_origin_from_apt_sources(target)
+        os_origin_apt_sources = self.apt_source_codename
         if self.current_os_release >= target and os_origin_apt_sources >= target:
```

Comment:
At this point, the extra variable seems unnecessary
```suggestion
        if self.current_os_release >= target and self.apt_source_codename >= target:
```